chain/cosmos/cosmos_chain.go

```diff
@@ -1263,83 +1300,21 @@ func (c *CosmosChain) StartConsumer(testName string, ctx context.Context, additi
 		}
 	}
 
-	providerHeight, err := c.Provider.Height(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to query provider height")
-	}
-	providerHeightInt64 := int64(providerHeight)
-
-	block, err := c.Provider.getFullNode().Client.Block(ctx, &providerHeightInt64)
-	if err != nil {
-		return fmt.Errorf("failed to query provider block to initialize consumer client")
-	}
-
 	genbz, err := validator0.GenesisFileContent(ctx)
 	if err != nil {
 		return err
 	}
 
-	// populate genesis file ccvconsumer module app_state.
-	// fetch provider latest block (timestamp, root.hash, and next_validators_hash) to populate provider_consensus_state
-	// populate provider_client_state with trusting and unbonding periods, latest_height.revision_height of height which is used for consensus state
-	// populate initial_val_set with provider val pubkeys and power
-
-	nextValidatorsHash := block.Block.NextValidatorsHash
-	timestamp := block.Block.Time
-	rootHash := block.Block.AppHash
-
-	page := int(1)
-	perPage := int(1000)
-	providerVals, err := c.Provider.getFullNode().Client.Validators(ctx, &providerHeightInt64, &page, &perPage)
+	ccvStateMarshaled, _, err := c.Provider.GetNode().ExecQuery(ctx, "provider", "consumer-genesis", c.cfg.ChainID)
 	if err != nil {
-		return fmt.Errorf("failed to get provider validators: %w", err)
+		return fmt.Errorf("failed to query provider for ccv state: %w", err)
 	}
-
-	initialVals := make([]abcitypes.ValidatorUpdate, len(providerVals.Validators))
-	for i, val := range providerVals.Validators {
-		initialVals[i] = abcitypes.ValidatorUpdate{
-			PubKey: crypto.PublicKey{Sum: &crypto.PublicKey_Ed25519{Ed25519: val.PubKey.Bytes()}},
-			Power:  val.VotingPower,
-		}
-	}
-
-	providerCfg := c.Provider.Config()
-
-	clientState := ibctmtypes.NewClientState(
-		providerCfg.ChainID,
-		ibctmtypes.DefaultTrustLevel,
-		DefaultProviderUnbondingPeriod/2,
-		DefaultProviderUnbondingPeriod, // Needs to match provider unbonding period
-		ccvprovidertypes.DefaultMaxClockDrift,
-		clienttypes.Height{
-			RevisionHeight: uint64(providerHeight),
-			RevisionNumber: clienttypes.ParseChainID(providerCfg.ChainID),
-		},
-		commitmenttypes.GetSDKSpecs(),
-		defaultUpgradePath,
-	)
-
-	root := commitmenttypes.MerkleRoot{
-		Hash: rootHash,
-	}
-
-	consensusState := ibctmtypes.NewConsensusState(timestamp, root, nextValidatorsHash)
-
-	ccvState := ccvconsumertypes.NewInitialGenesisState(
-		clientState,
-		consensusState,
-		initialVals,
-		ccvconsumertypes.DefaultGenesisState().GetParams(),
-	)
-
-	ccvState.Params.Enabled = true
-
-	ccvStateMarshaled, err := c.cfg.EncodingConfig.Codec.MarshalJSON(ccvState)
-	c.log.Info("HERE STATE!", zap.String("GEN", string(ccvStateMarshaled)))
+	consumerICS := c.GetNode().ICSVersion(ctx)
+	providerICS := c.Provider.GetNode().ICSVersion(ctx)
+	ccvStateMarshaled, err = c.transformCCVState(ctx, ccvStateMarshaled, consumerICS, providerICS)
```

Comment:
I do like how this abstracts out the setup. My worry though is that as ICS upgrades to newer versions, teams using this will have to get upstream patches in vs being able to fix in their own test. May not be an issue, we just need to keep refernce of the old way to do it as well

Reply:
yeah, i think this'll depend on how ICS breaks as it upgrades to other versions; ideally the basic flow remains the same, and maybe we have to keep track of how the transform exactly happens